### ISSUE TITLE
Running Traefik with Docker 1.12 Swarm Mode

### ISSUE THREAD

**Issue by timchunght** [containous/traefik#504]
What I am attempting to do is run Traefik as a `docker service` in the new swarm mode and have it auto discover all the other services.
```
docker service create --name traefik -p 80:80 -p 8080:8080 traefik 
```
However, without the ability to bind the docker socket in swarm mode, what can we do to have traefik do auto service discovery for all the other services in docker swarm. Thanks

**Comment by timchunght** [comment]
My goal is to run traefik as a global service in docker swarm and have it communicate with the docker swarm API and use the `label` of the `service` (this use case is unique the docker 1.12) in docker swarm for hostname matching.

https://docs.docker.com/engine/reference/commandline/service_create/

**Comment by emilevauge** [comment]
@timchunght for sure, we will make traefik compliant with Docker swarm mode :)

**Comment by vdemeester** [comment]
We are watching closely the 1.12 feature and how they will evolve between 1.12 and 1.13, but we're definitely going to support that :angel: :wink:

**Comment by katopz** [comment]
WoW, Can't wait! 👯 🎵

**Comment by timchunght** [comment]
@vdemeester is there any timeline on this feature. Really excited and looking forward to it! Thanks!

**Comment by PanJ** [comment]
I'm new to traefik but searching for reverse proxy solution in docker swarm mode. I think, in the meantime, we can use file backend and route them using domain name with attached network. Swarm already handles service discovery and load balancing.

**Comment by diegofernandes** [comment]
I`m writing a code to support that, let`s what happen.

**Comment by emilevauge** [comment]
Fixed by #602

**Comment by Regner** [comment]
Think you mean fixed by #602 ;)
